syntax = "proto3";

package iam;

service IAMService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse);
  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);

  rpc CheckAdmin(AdminCheckRequest) returns (AdminCheckResponse);
  rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse);
  rpc BanUser(BanUserRequest) returns (BanUserResponse);
}

message GetUserRequest {
  int64 user_id = 1;
}
message GetUserResponse {
  int64 id = 1;
  string name = 2;
  string email = 3;
  string city = 4;
  bool is_active = 5;
  string avatar_url = 6;
}

message UpdateUserRequest {
  int64 user_id = 1;
  map<string,string> fields = 2;
}
message UpdateUserResponse {
  bool ok = 1;
}

message AdminCheckRequest {
  int32 admin_id = 1;
}

message AdminCheckResponse {
  int32 id = 1;
  string email = 2;
  bool is_admin = 3;
}

message DeleteUserRequest {
  int32 user_id = 1;
}

message DeleteUserResponse {
  bool success = 1;
}

message BanUserRequest {
  int32 user_id = 1;
}

message BanUserResponse {
  bool success = 1;
}